tests/test_db_service.py

```diff
@@ -161,45 +138,62 @@ async def test_db_operation_is_retried_with_increasing_interval(self):
+        self.collection.insert_one.return_value = InsertOneResult(ObjectId("0000000000aa1111111111bb"), True)
         folder = await self.test_service.create("folder", self.folder_stub)
         self.collection.insert_one.assert_called_once_with(self.folder_stub)
         self.assertTrue(folder)
 
     async def test_read_folder_returns_data(self):
         """Test that read method works for folder and returns folder."""
-        self.collection.find_one.return_value = futurized(self.folder_stub)
+        self.collection.find_one.return_value = self.folder_stub
         found_folder = await self.test_service.read("folder", self.f_id_stub)
         self.collection.find_one.assert_called_once_with({"folderId": self.f_id_stub}, {"_id": False})
         self.assertEqual(found_folder, self.folder_stub)
 
-    async def test_eppn_exists_returns_false(self):
-        """Test that eppn exists method works and returns None."""
-        found_doc = await self.test_service.exists_eppn_user("test_user@eppn.fi", "name")
+    async def test_published_folder_returns_data(self):
+        """Test that published folder checks if folder is published."""
+        self.collection.find_one.return_value = self.folder_stub
+        found_folder = await self.test_service.published_folder(self.f_id_stub)
+        self.collection.find_one.assert_called_once_with(
+            {"published": True, "folderId": self.f_id_stub}, {"_id": False}
+        )
+        self.assertEqual(found_folder, True)
+
+    async def test_externalId_exists_returns_false(self):
+        """Test that externalId exists method works and returns None."""
+        self.collection.find_one.return_value = None
+        found_doc = await self.test_service.exists_user_by_externalId("test_user@eppn.fi", "name")
         self.assertEqual(found_doc, None)
         self.collection.find_one.assert_called_once_with(
-            {"eppn": "test_user@eppn.fi", "name": "name"}, {"_id": False, "eppn": False}
+            {"externalId": "test_user@eppn.fi", "name": "name"}, {"_id": False, "externalId": False}
         )
 
-    async def test_eppn_exists_returns_true(self):
-        """Test that eppn exists method works and returns user id."""
-        self.collection.find_one.return_value = futurized(self.user_stub)
-        found_doc = await self.test_service.exists_eppn_user("test_user@eppn.fi", "name")
+    async def test_externalId_exists_returns_true(self):
+        """Test that externalId exists method works and returns user id."""
+        self.collection.find_one.return_value = self.user_stub
+        found_doc = await self.test_service.exists_user_by_externalId("test_user@eppn.fi", "name")
         self.assertEqual(found_doc, self.user_id_stub)
         self.collection.find_one.assert_called_once_with(
-            {"eppn": "test_user@eppn.fi", "name": "name"}, {"_id": False, "eppn": False}
+            {"externalId": "test_user@eppn.fi", "name": "name"}, {"_id": False, "externalId": False}
```

Comment:
`externalId` functions and variables should be `external_id`

Reply:
variables yes, JSON key values should be camelCase